evals/agent_evals/agent_testing_utils.py

```diff
@@ -0,0 +1,103 @@
+import nbformat
+import sys
+import os
+
+
+from evals.prompts.constants import JUPYTER_NOTEBOOK_SECTION_HEADING, VARIABLES_SECTION_HEADING, FILES_SECTION_HEADING, USER_TASK_HEADING_SECTION
+
+
+def process_notebook_update(notebook, cell_update):
+    """Create the output notebook based on the specified cell update"""
+    cell_type = cell_update["cell_type"]
+    code = cell_update["code"]
+    update_type = cell_update["type"]
+
+    new_cell = nbformat.v4.new_code_cell(source=code) if cell_type == "code" else nbformat.v4.new_markdown_cell(
+        source=code)
+
+    if update_type == "new":
+        index = cell_update["index"]
+        notebook.cells.insert(index, new_cell)
+    elif update_type == "modification":
+        target_cell_id = cell_update["id"]
+        for cell in notebook.cells:
+            if cell.get("id") == target_cell_id:
+                cell["source"] = code
+                break
+    else:
+        raise ValueError(f"Unsupported update type: {update_type}")
+
+    all_code = "\n\n".join(
+        cell.source for cell in notebook.cells if cell.cell_type == "code"
+    )
+    # print(all_code)
```

Comment:
Before we merge this in, let's just get rid of these commented out print statements across the PR